pkg/trace/agent/normalizer.go

```diff
@@ -152,6 +152,12 @@ func normalizeChunk(chunk *pb.TraceChunk, root *pb.Span) {
 		if p, ok := root.Metrics[tagSamplingPriority]; ok {
 			chunk.Priority = int32(p)
 		}
+		for _, s := range chunk.Spans {
```

Comment:
Can we maybe find another place where we already iterate over the whole chunk? Also, is there any way to avoid the loop is let’s say chunk.Priority != 0 ?